content/shared/tutorials/_testing_push_android.textile

```diff
@@ -0,0 +1,101 @@
+      return;
+    }
+    console.log('Push notification published');
+    res.send("Push Sent");
+  });
+})
+```
+
+Restart the server and navigate to @YOUR-SERVER-URL/push/device@ in your browser. You should be able to receive a notification on your device as shown the image below.
+
+<a href="/images/tutorials/android-push/android-push-screenshot.jpg" target="_blank">
+	 <img src="/images/tutorials/android-push/android-push-screenshot.jpg" style="width: 50%; height: 50%" alt="Android Push Notification Screenshot">
+</a>
+
+h3. b. Testing using Client ID
+
+We can test sending push notifications using the client ID, this allows us to push all devices with that client ID by adding the following code to your index.js file:
+
+```[nodejs]
+app.get('/push/client', function (req, res) {
+  var recipient = {
+    clientId: 'iosPushClient'
+  };
+  var notification = {
+    notification: {
+      title: 'Hello from Ably!'
+    }
+  };
+  realtime.push.publish(recipient, notification, function(err) {
+    if (err) {
+      console.log('Unable to publish push notification; err = ' + err.message);
+      return;
+    }
+    console.log('Push notification published');
+    res.send("Push Sent");
+  });
+})
+```
+Note: If your clientId is different you will also need to change this.
+
+Restart the server and go to @YOUR-SERVER-URL/push/client@ you should receive a notification on your device as shown in the image below.
+
+<a href="/images/tutorials/android-push/android-push-screenshot.jpg" target="_blank">
+	 <img src="/images/tutorials/android-push/android-push-screenshot.jpg" style="width: 50%; height: 50%" alt="Android Push Notification Screenshot">
+</a>
+
+h3. c. Testing using Channels
+
+Channel based Push Notifications allow us to push messages to a channel that the client on the device has subscribed to. To test this, add the following code to our index.js:
+
+```[nodejs]
+app.get('/push/channel', function (req, res) {
+  var extras = {
+    push: {
+      notification: {
+        title: 'Hello from Ably!',
+        body: 'Example push notification from Ably.'
+      },
+      data: {
+        foo: 'bar',
+        baz: 'qux'
+      }
+    }
+  };
+
+  var channel = realtime.channels.get('push');
+  channel.publish({ name: 'example', data: 'data', extras: extras }, function(err) {
+    if (err) {
+      console.log('Unable to publish message with push notification; err = ' + err.message);
+      return;
+    }
+    console.log('Message with push notification published');
+    res.send("Push Sent");
+  });
+})
+```
+
+Restart the server and go to @YOUR-SERVER-URL/push/channel@, you should receive a notification on your device. Notice how this notification has both a title and a body. You can do this on all three types of push notifications mentioned earlier, if you wish. Simply add a "body" 'key/value' pair to your notification part of the payload. See the above code for an example of this.
```

Comment:
`Restart the server and go to @YOUR-SERVER-URL/push/channel@, you should receive... `. The comma here feels quite stilted, would be good to restructure or replace with a full stop.

Reply:
`Simply add a "body" 'key/value' pair.. `. `See the above code for an example of this.`. This may just be not being 100% on push, but in the above `body` seems to be `'Example push notification from Ably.'`. is this a `key/value` pair?